core/src/main/java/hudson/model/Queue.java

```diff
@@ -810,17 +809,18 @@ private WaitingItem peek() {
             r = checkPermissionsAndAddToList(r, p);
         }
         for (BuildableItem p : reverse(s.pendings)) {
-            r= checkPermissionsAndAddToList(r, p);
+            r = checkPermissionsAndAddToList(r, p);
         }
         Item[] items = new Item[r.size()];
         r.toArray(items);
         return items;
     }
 
     private List<Item> checkPermissionsAndAddToList(List<Item> r, Item t) {
-        if (t.task instanceof hudson.security.AccessControlled) {
-            if (((hudson.security.AccessControlled)t.task).hasPermission(hudson.model.Item.READ)
-                    || ((hudson.security.AccessControlled) t.task).hasPermission(hudson.security.Permission.READ)) {
+        if (t.task instanceof AccessControlled) {
+            AccessControlled taskAC = (AccessControlled) t.task;
+            if (taskAC.hasPermission(hudson.model.Item.READ)
```

Comment:
any reason hudson.model.Item isn't imported?

Reply:
Due to Queue.Item defined in the class :)